d3rlpy/base.py

```diff
@@ -83,6 +87,8 @@ def __init__(self, batch_size, n_frames, scaler, augmentation, use_gpu):
 
         self.impl = None
         self.eval_results_ = {}
+        self.loss_history_ = {}
```

Comment:
Since you add `defaultdict` import here, you can replace these with `defaultdict(list)`

Reply:
Yes that was the intention, I have observed the usage pattern and it seems more adequate to use defaultdict(list) for those things. However testing it I received some errores and though it was due to defaultdict not being able to be jason friendly. At the issue is a problem with the reload method of Ipython that seems to have difficulties with the reference to Impl.

So I will switch to defaultdict(list) as soon I test it.

Reply:
Oh, I see. Maybe  we can explicitly omit the private properties here.
https://github.com/takuseno/d3rlpy/blob/f04b303d976a03588f15655757458a0848d574b0/d3rlpy/base.py#L430